processor/formatters.go

```diff
@@ -284,6 +283,122 @@ func toCSV(input chan *FileJob) string {
 	return b.String()
 }
 
+func toHtml(input chan *FileJob) string {
+	return `<html lang="en"><head><meta charset="utf-8" /><title>scc html output</title><style>table { border-collapse: collapse; }td, th { border: 1px solid #999; padding: 0.5rem; text-align: left;}</style></head><body>` +
+		toHtmlTable(input) +
+		`</body></html>`
+}
+
+func toHtmlTable(input chan *FileJob) string {
+	languages := map[string]LanguageSummary{}
+	var sumFiles, sumLines, sumCode, sumComment, sumBlank, sumComplexity int64 = 0, 0, 0, 0, 0, 0
+
+	for res := range input {
+		sumFiles++
+		sumLines += res.Lines
+		sumCode += res.Code
+		sumComment += res.Comment
+		sumBlank += res.Blank
+		sumComplexity += res.Complexity
+
+		_, ok := languages[res.Language]
+
+		if !ok {
+			files := []*FileJob{}
+			files = append(files, res)
+
+			languages[res.Language] = LanguageSummary{
+				Name:       res.Language,
+				Lines:      res.Lines,
+				Code:       res.Code,
+				Comment:    res.Comment,
+				Blank:      res.Blank,
+				Complexity: res.Complexity,
+				Count:      1,
+				Files:      files,
+			}
+		} else {
+			tmp := languages[res.Language]
+			files := append(tmp.Files, res)
+
+			languages[res.Language] = LanguageSummary{
+				Name:       res.Language,
+				Lines:      tmp.Lines + res.Lines,
+				Code:       tmp.Code + res.Code,
+				Comment:    tmp.Comment + res.Comment,
+				Blank:      tmp.Blank + res.Blank,
+				Complexity: tmp.Complexity + res.Complexity,
+				Count:      tmp.Count + 1,
+				Files:      files,
+			}
+		}
+	}
+
+	language := []LanguageSummary{}
+	for _, summary := range languages {
+		language = append(language, summary)
+	}
+
+	language = sortLanguageSummary(language)
```

Comment:
Looks like this code to summarise the results is duplicated across multiple formatters - maybe it can be lifted out to a shared function?

Reply:
Yep it is. Ill merge then refactor it a bit later.